diff --git a/src/getargs.c b/src/getargs.c
index ef559296..7da2363c 100644
--- a/src/getargs.c
+++ b/src/getargs.c
@@ -896,16 +896,17 @@ getargs (int argc, char *argv[])
       }
   }
 
-  if (argc - optind != 1)
+  uint32_t optind32 = (uint32_t)optind;
+  if (argc - optind32 != 1)
     {
-      if (argc - optind < 1)
+      if (argc - optind32 < 1)
         error (0, 0, _("missing operand"));
       else
-        error (0, 0, _("extra operand %s"), quote (argv[optind + 1]));
+        error (0, 0, _("extra operand %s"), quote (argv[optind32 + 1]));
       usage (EXIT_FAILURE);
     }
 
-  grammar_file = uniqstr_new (argv[optind]);
+  grammar_file = uniqstr_new (argv[optind32]);
   MUSCLE_INSERT_C_STRING ("file_name", grammar_file);
 }
 
